src/main.go

```diff
@@ -120,6 +118,12 @@ func main() {
 	if success {
 		fmt.Println("\nGGWP, you solved the puzzle!")
 	}
+
+	close(interruptChan)
```

Comment:
You should `defer` the close. If an error occurs before reaching this point, your channel will never be closed.